Cargo.toml

```diff
@@ -27,6 +28,7 @@ tokio-tls = "0.1"
 url = "1.2"
 uuid = { version = "0.5", features = ["v4"] }
 hyper-proxy = "0.4.0"
+uchardet = "2.0"
```

Comment:
Can't use [chardet](https://crates.io/crates/chardet) because of [license issue](https://github.com/thuleqaid/rust-chardet/issues/3), it's [LGPL-3.0](https://github.com/thuleqaid/rust-chardet/blob/master/LICENSE)